### PR TITLE
Create daily database population schedule

### PR DESCRIPTION
**Pull request by kidzegeye**
## Overview

Made a cronjob that populates the model data every hour within the school day

## Changes Made

- Create a `populate_models` command that populates the data of for each model in the same way that the `\cdn\populate\` endpoint does
- Create `update_db.txt` which schedules a cron job to run `populate_models`
- Modify the docker-compose files to run the cron on startup

## Test Coverage

Tested locally by running `populate_models` on its own and by running `docker-compose up` to see how the cronjob would behave in the container

## Next Steps (delete if not applicable)

Possibly delete the `\cdn\populate\` endpoint, as populating the tables takes too long, which causes the server to timeout

### CODE REVIEW

**Review by mateow99 — APPROVED** [review]
lgtm

File: Dockerfile [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
 WORKDIR /usr/app
 COPY ./src .
 COPY ./requirements.txt .
-RUN pip install -r requirements.txt
\ No newline at end of file
+RUN pip install -r requirements.txt
+
+COPY update_db.txt /etc/cron.d/update_db
+RUN chmod 0644 /etc/cron.d/update_db
+RUN chmod 0744 manage.py
```
Comment (mateow99) [inline Dockerfile]: can you explain why these numbers? is there any reason to restrict privileges

**Review by mateow99 — APPROVED** [review]
lgtm but why did we need to switch from scheduler to cronjob?

File: src/eatery/controllers/populate_eatery.py [hunk 2]
```diff
@@ -42,8 +42,8 @@ def generate_eatery(self, json_eatery):
         eatery = EaterySerializer(data=data)
         if eatery.is_valid():
             eatery.save()
-        else:
-            print(eatery.errors)
+        #else:
```
Comment (snajima) [inline src/eatery/controllers/populate_eatery.py:45]: let's not leave in commented code

File: docker-compose.yml [hunk 3]
```diff
@@ -4,7 +4,9 @@ services:
   app:
     build: .
     env_file: .env
-    command: sh -c "python manage.py migrate && python manage.py runserver 0.0.0.0:80"
+    stdin_open: true # docker run -i
+    tty: true # docker run -t
+    command: sh -c "python manage.py migrate && printenv > /etc/environment && { cron -f & python manage.py runserver 0.0.0.0:8; }"
```
Comment (snajima) [inline docker-compose.yml]: why are we running on port 8? 

also, consider changing `/etc/environment` to `/etc/profile.d/`: https://askubuntu.com/questions/866161/setting-path-variable-in-etc-environment-vs-profile

**Review by snajima — CHANGES_REQUESTED** [review]
j some few nits here and there :)

File: docker-compose.yml (hunk 3 above)
Comment (kidzegeye) [reply docker-compose.yml]: Will the cronjob still be able to access the env vars if the environment variables are in /etc/profile.d/?

File: Dockerfile (hunk 1 above)
Comment (kidzegeye) [reply Dockerfile]: I found the crontab file in the old eatery repo, so I thought it would be better to use cron than a external library

Comment (kidzegeye) [reply Dockerfile]: > can you explain why these numbers? is there any reason to restrict privileges

Fixed

File: docker-compose.yml (hunk 3 above)
Comment (snajima) [reply docker-compose.yml]: yeah, i don't think it's good to directly pipe into `/etc/environment`

instead pipe the output of `printenv | awk '{print "export " $0}'` into a `.sh` file in `/etc/profile.d/` (executed every shell initialization)

**Review by snajima — APPROVED** [review]
just change the `docker-compose.yml` issue addressed above and then we can get this merged :)

**Pull request merged**